bliss/train.py

```diff
@@ -23,6 +23,7 @@ def train(cfg: DictConfig):  # pylint: disable=too-many-branches, too-many-state
 
     # setup dataset
     simulator_cfg = cfg.cached_simulator if cfg.training.use_cached_simulator else cfg.simulator
+    simulator_cfg = cfg.surveys.dc2 if cfg.training.use_dc2 else simulator_cfg
```

Comment:
Are we training the encoder on real DC2 data and not simulated DC2 data?

Reply:
I see. Since DC2 is simulated data, perhaps it would be good for `DC2` class to also inherit from `CachedSimulatedDataset`, and somehow set `CachedSimulatedDataset.cached_data_path` to be DC2's `survey_data_dir`.

Reply:
That's not a code that `DC2` can reuse from `CacheSimulatedDataset`.
But `DC2` should inherit from the parent classes of `CacheSimulatedDataset`? i.e., `pl.LightningDataModule` and `Dataset`?

Reply:
Oh never mind, I see `Survey` already inherits from these two classes.

Reply:
See my comments above. If we replace `use_cached_simulator` and `use_dc2` with a new field called `data_source`, I think these two lines of can go away. Instead, we'd instantiate `cfg.training.data_source`, whatever it's set to.

Reply:
this could work; but see https://github.com/prob-ml/bliss/pull/885#discussion_r1268872096

Reply:
@XinyueLi1012 reading the file as something other than .pt can possibly be made more general; I just think DC2 should be a `CachedSimulatedDataset` to stay as close as possible to BLISS' code pattern

Reply:
Is the transformation (to .pt format that BLISS uses) very cumbersome? Also, are there other difficulties with inheriting from `CachedSimulatedDataset`?

Reply:
For SDSS and DECaLS, we don't convert the survey data to `.pt` files. DC2 isn't exactly like these other surveys, in that it has a ground truth that we can use for training, but it is a lot like them; therefore, it's not obvious to me why DC2 should be converted to `.pt` before processing, while we don't require the same of the other surveys.

Not that it would be a problem to do so necessarily. The main drawback I see is that we'd need an additional 5 TB of disk space to store the converted version of DC2. Also, it would add another step, so a bit more complicated from a user's perspective.

Reply:
I think for ease of implementation, we can use this `data_source` pattern. But eventually I think it'd be best to generalize the data loading to also handle non-`.pt` disk dataset files.